contracts/tequila-train/io/src/lib.rs

```diff
@@ -146,17 +138,39 @@ pub enum Command {
     ),
 }
 
+#[derive(Encode, Decode, TypeInfo, Clone, Debug)]
+pub enum Event {
+    Skipped,
+    Placed {
+        tile_id: u32,
+        track_id: u32,
+        remove_train: bool,
+    },
+    Registered {
+        player: ActorId,
+        name: String,
+    },
+    GameStarted,
+    GameRestarted(
+        /// Optional players limit.
+        Option<u64>,
+    ),
```

Comment:
In my opinion, it is better to use struct rather than tuple if inner type is not evident.
```suggestion
    GameRestarted {
        players_limit: Option<u64>,
    },
```